src/formatDate.js

```diff
@@ -9,6 +9,81 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  let newDate = '';
+  let fromCharacter = 0;
+  let month;
+  let day;
+  let year;
+  let yearFormat;
+
+  fromFormat.forEach((element) => {
+    const toCharacter = fromCharacter + element.length;
+    const separator = fromFormat[fromFormat.length - 1];
+    const currentCharacter = fromCharacter + element.length + separator.length;
+
+    if (element === 'MM') {
+      month = date.slice(fromCharacter, toCharacter);
+      fromCharacter = currentCharacter;
+    }
+
+    if (element === 'DD') {
+      day = date.slice(fromCharacter, toCharacter);
+      fromCharacter = currentCharacter;
+    }
+
+    if (element === 'YYYY') {
+      year = date.slice(fromCharacter, toCharacter);
+      fromCharacter = currentCharacter;
+      yearFormat = element;
+    }
+
+    if (element === 'YY') {
+      year = date.slice(fromCharacter, toCharacter);
+      fromCharacter = currentCharacter;
+      yearFormat = element;
+    }
+  });
+
+  toFormat.forEach((element, index) => {
+    if (element === 'DD') {
+      newDate += day;
+    }
+
+    if (element === 'MM') {
+      newDate += month;
+    }
+
+    if (element === 'YY') {
+      if (yearFormat === 'YY') {
+        newDate += year;
+      } else if (yearFormat === 'YYYY') {
+        newDate += year.slice(-2);
+      }
+    }
+
+    if (element === 'YYYY') {
+      if (yearFormat === 'YYYY') {
+        newDate += year;
+      } else if (yearFormat === 'YY') {
+        if (year < '30') {
```

Comment:
When converting from `YY` to `YYYY`, the comparison `year < '30'` is a string comparison rather than a numeric comparison. This could lead to incorrect behavior if `year` is not explicitly converted to a number before the comparison.